### ISSUE TITLE
certain playlists not syncing completely

### ISSUE THREAD

**Issue by simon-weber** [simon-weber/Autoplaylists-for-Google-Music#193]
This consolidates multiple older issues (https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/186, https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/185), all of which describe these symptoms:

* some playlists contain less songs than they should, as shown through the "test" button
* creating a new autoplaylist successfully creates a playlist on Google's end (unlike https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/192, where it does not)
* affected playlists may grow smaller over time as syncs occur
* recreating the playlist (eg with the "duplicate" button) may allow it to grow larger
* the logs show the correct songs being added, but being rejected with a TOO_MANY_ITEMS response code 

In the past, I'd only seen TOO_MANY_ITEMS when attempting to add more than 1k entries to a playlist. I'm not sure why a bunch of users are seeing it now, with playlists often far smaller than 1k.

My best lead is that Google's clients don't seem affected by playlists in this state. On a quick look I didn't notice anything obviously different about my requests vs theirs, but it's possible there's something subtle (or even another api involved).

**Comment by johnrobertlawson** [comment]
Do you reckon this is anything to do with the new Youtube Music service and potential back-end change-ups?

**Comment by simon-weber** [comment]
Yeah, I think that's the most likely explanation right now, especially with how there were very sudden problems a) starting mid-day pacific time and b) without changes on our end.

**Comment by johnrobertlawson** [comment]
Well, I just saw Soundiiz (online playlist sync service) say that Google have recently implemented a limit of 10 playlists created a day on YouTube (music?). I am not sure about your implementation, but does Google "see" new playlists being created when the extension is updating the track list? Soundiiz said they are waiting for more information from Google. I can't find their "notification" about this issue - it just popped up and disappeared on their website.

**Comment by simon-weber** [comment]
It shouldn't see a new playlist except for when a new autoplaylist is created. That definitely sounds like it could be related, though (especially if they also limited the number of updates to a playlist in addition to the number of creations).

**Comment by simon-weber** [comment]
How's everyone's syncing working? My metrics have looked pretty good lately.

**Comment by brianzam** [comment]
Weird.  Library seems to be a bit corrupted, some songs appearing twice in autoplaylists.  Generally better with restricting syncs but still disparity between songs autoplaylists says are in the playlist and songs that appear on google play music.

Have re-set/repait library in settings.

One song was added by simply removing it from my library and adding it again. Others take more work.  It's down to managing every song that is an exception now.  Not sure I have the time or the inclination!

Brian

**Comment by simon-weber** [comment]
Hm. I could add a section to the syncing page about discrepancies, if being able to detect them more easily would be helpful. That's essentially what syncing is doing already; the work would be more about just presenting the diff in a meaningful way.

Are these songs subscription tracks, or one's that you've uploaded? If they're subscription tracks it's something I could potentially automate.

**Comment by Noggog** [comment]
Still having syncing issues.  Maybe I'm on an older version though.  Same stuff.  900 song playlist slowly widdling down every time to less and less.

I've also noticed that it's not just with the massive playlists that are piping in the whole library.  It also happens with the smaller ones, too.  I have "showcase" playlist with some hand picked songs, except I put them on a few separate lists by genre.  I then have the autoplaylist tool combining those lists into one master one.  The result should be about ~84 songs, but it ends up with 78 as of today.

**Comment by brianzam** [comment]
Hi Simon,

I think it affects both, but mainly the subscription.  Also think that it may be related to YouTube playlists as many of the songs I’m having trouble with have the ‘View Video’ right click option.

Some kind of discrepancy log / user debug section might be helpful, thanks – even if you give us the raw feed so we can send it back to you. See if you can highlight any ‘problem’ discrepancies?

Brian

**Comment by simon-weber** [comment]
Hm. How many autoplaylists do you folks have, and how many total tracks in them?

The subscription/youtube lead is an interesting one. My day-to-day personal account is a free one, but I've got a subscription test account I can use to see if there's anything odd about those tracks. Way back in the day I did have some issues with syncing subscription tracks (since they use different ids in different cases), but I'd be surprised if that was related; those symptoms were more like tracks never getting into playlists.

**Comment by Noggog** [comment]
I have 8 playlists.

7 of them are different filtered "windows" on my whole library, with 900 as their size. 
One of them is just merging 3 other playlists, with a total of about ~85 songs.

Obviously, the problem stated means none of these have the listed numbers above, but that's what they're trying to get.

Lately, too, I haven't been automatically syncing.  I just sync single playlists by hand, so I doubt it's an overloading thing, especially since the problem shows on my 85 song playlist

**Comment by brianzam** [comment]
6 playlists for me, max no. of tracks about 700.

**Comment by brianzam** [comment]
Well now.  Not quite sure what's changed.  I have known for a while that some playlists need to be synched several times before the song order is right (it increments - can explain later).  So I thought I would just delete the playlist in Google Music then autoplaylist would re-add it.  Appreciate that was a bit rash.  The playlist simply disappeared from Google Music and no amount of synching would bring it back.

No worries, duplicated the original playlist in autoplaylists and all is good.

Then noticed that the track numbers now match exactly, in all playlists, not just the one I messed around with. May just be a coincidence, but go figure!

On a side note, much better approach to re-building a playlist on Google Music is to delete all but one song in Google Music (if I delete all songs autoplaylist can go a bit screwy - sorry Simon!), then simply re-save the playlist in autoplaylist - works like a charm.  I'd suggest to others who are having the issue with the track numbers not matching could give that a go.

Hope this helps.

**Comment by simon-weber** [comment]
Ah, yeah, deleting an existing playlist will lead to https://github.com/simon-weber/Autoplaylists-for-Google-Music/issues/160. Duplicating it is the right move to fix it.

That's interesting that your syncing seems to be working again afterwards. I'd imagine it's a coincidence - I haven't noticed that problem in any of the other example logs - but maybe there's something to it that I haven't considered yet.
